src/main/java/uk/gov/ida/verifyserviceprovider/configuration/VerifyServiceProviderConfiguration.java

```diff
@@ -39,7 +37,7 @@ public VerifyServiceProviderConfiguration(
             @JsonProperty("samlSigningKey") @NotNull @Valid @JsonDeserialize(using = PrivateKeyDeserializer.class) PrivateKey samlSigningKey,
             @JsonProperty("samlPrimaryEncryptionKey") @NotNull @Valid @JsonDeserialize(using = PrivateKeyDeserializer.class) PrivateKey samlPrimaryEncryptionKey,
             @JsonProperty("samlSecondaryEncryptionKey") @Valid @JsonDeserialize(using = PrivateKeyDeserializer.class) PrivateKey samlSecondaryEncryptionKey,
-            @JsonProperty("msaMetadata") @NotNull @UnwrapValidatedValue @Valid Optional<MsaMetadataConfiguration> msaMetadata,
+            @JsonProperty("msaMetadata") @NotNull Optional<@NotNull @Valid MsaMetadataConfiguration> msaMetadata,
```

Comment:
My head is struggling to understand what's happening here?

Reply:
Me too.  Before @UnwrapValidatedValue extracted a thing from a wrapper (like Optional) and then did the validation on the extracted value.  That annotation disappeared.  As I understand it, you can now put your annotations next to the wrapped type and it all just works.  At first I took out the original @NotNull but figured the optional value should not be null as well as the wrapped value so we're making sure there's a value for that and then validating the contents.